test/test_csvs.py

```diff
@@ -0,0 +1,291 @@
+                # the trend fields
+                elif header[index] in ('TrendTest_TimeIntLengthDays','TrendTest_PolynomialOrder','TrendTest_nstd'):
+                    if not self._is_empty_or_numeric(part):
+                        invalid.append(header[index] + ': [' + part + ']')
+
+        return invalid
+
+    # ###########################################
+    # verify the files needed for the tests exist
+    # ###########################################
+
+    # the PSVs are built from the CSVs in setuptest.sh to make them pipe-separated files
+    # due to the "local" file using commas inside the fields of a handful of records
+
+    def test_globalFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_global_range_values.psv')))
+
+    def test_gradientFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_gradient_test_values.psv')))
+
+    def test_localFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_local_range_values.psv')))
+
+    def test_spikeFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_spike_test_values.psv')))
+
+    def test_stuckFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_stuck_test_values.psv')))
+
+    def test_trendFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_trend_test_values.psv')))
+
+    # these files are generated within setuptest.sh from the asset-management
+    # and preload-database repositories
+
+    def test_parameterFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('parameter.txt')))
+        parameters = self._get_parameters()
+        self.assertTrue(len(parameters) > 0)
+
+    def test_refdesFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('refdes.txt')))
+        refdes = self._get_ref_designators()
+        self.assertTrue(len(refdes) > 0)
+
+    def test_typeFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('type.txt')))
+        types = self._get_types()
+        self.assertTrue(len(types) > 0)
+
+    def test_unitFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('unit.txt')))
+        units = self._get_units()
+        self.assertTrue(len(units) > 0)
+
+    def test_validateGlobalRangeFile(self):
+        validfile = True # file is valid until proven otherwise
+        header = None # obtained from 1st record
+        # the numbers of the fields in each data record needing validation
+        flds_to_validate = None # established from 1st record
+
+        with io.open(self._getpath('data_qc_global_range_values.psv'),mode='rt',encoding='iso-8859-1') as f:
+            for index,l in enumerate(f):
+                line = l.strip() # drop the trailing newline
+                # the 1st record contains the headers
```

Comment:
Use csv, not `split`

Reply:
I'm using pandas, although csv appears to be a good option as well